apps/dashboard/src/components/primitives/tag-input.tsx

```diff
@@ -52,21 +79,37 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
   };
 
   return (
-    <div className="flex flex-col space-y-2">
-      <InputField>
-        <Input
-          ref={ref}
-          type="text"
-          value={inputValue}
-          onKeyDown={handleKeyDown}
-          className={cn('flex-grow', className)}
-          placeholder="Type a tag and press Enter"
-          onChange={(e) => setInputValue(e.target.value)}
-          {...rest}
-        />
-      </InputField>
+    <div className="flex flex-col space-y-2" ref={divRef}>
+      {showAddButton && isFocussed && (
```

Comment:
@BiswaViraj, please speak to @desiprisg about this field, as there were some changes to how it should work 

---
the focused state could be handled with css and we can animate nicely the add button